### PR TITLE
docs: update daemon documentation

### PR DESCRIPTION
**Pull request by lucasmoura**
## Why is this needed?
Update the daemon documentation to state that it now also runs on Azure while also explaining the retry auto-attach mechanism the daemon is now running.

Fixes: #2922

- [ ] *(un)check this to re-run the checklist action*

### CODE REVIEW

File: docs/explanations/what_is_the_daemon.rst [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
 
-Ubuntu Pro Client sets up a daemon on supported platforms (currently GCP
-only) to detect if an Ubuntu Pro license has been purchased for the machine.
+Ubuntu Pro Client sets up a daemon for two types of machines:
+
+* Azure and GCP generic cloud images
+* Cloud Pro images
+
+For the first type of images, the daemon is responsible
+to detect if an Ubuntu Pro license has been purchased for the machine.
```
Comment (s-makin) [inline docs/explanations/what_is_the_daemon.rst:10]:
```suggestion
for detecting if an Ubuntu Pro license has been purchased for the machine.
```

File: docs/explanations/what_is_the_daemon.rst [hunk 2]
```diff
@@ -1,10 +1,39 @@
...
 If a Pro license is detected, then the machine is automatically attached.
 
+For the Pro cloud images, the daemon is responsible for retrying the auto-attach
+operation if it fails on first boot. When that happens, users will be notified in
+MOTD or `pro status` with a message with pattern:
```
Comment (s-makin) [inline docs/explanations/what_is_the_daemon.rst:15]:
```suggestion
MOTD or when they run `pro status` a message like:
```

File: docs/explanations/what_is_the_daemon.rst [hunk 3]
```diff
@@ -1,10 +1,39 @@
...
+
+.. code-block:: text
+
+   Failed to automatically attach to an Ubuntu Pro subscription 1 time
+   The failure was due to: ERROR_MSG
+   The next attempt is scheduled for RETRY_DATE
+   You can try manually with `sudo pro auto-attach`.
+
+The retry auto-attach operation will keep retrying for a **month**. If at the end, we cannot attach
```
Comment (s-makin) [inline docs/explanations/what_is_the_daemon.rst:24]:
```suggestion
The "retry auto-attach" operation will keep retrying for a **month**. If at the end, we cannot attach
```

Comment (s-makin) [reply docs/explanations/what_is_the_daemon.rst:24]: For a month, but at what frequency?

**Review by s-makin — COMMENTED** [review]
Just a few suggestions

File: docs/explanations/what_is_the_daemon.rst (hunk 3 above)
Comment (lucasmoura) [reply docs/explanations/what_is_the_daemon.rst:24]: It changes as time goes by. It starts retrying after 15min after a failure and every retry takes longer than the last attempt. Do you think it is worth explaining that ?

Comment (s-makin) [reply docs/explanations/what_is_the_daemon.rst]: I think probably not for now. The explanation might lead to more confusion or questions. Happy to leave it as is for now!

**Review by s-makin — APPROVED** [review]
LGTM, thanks :)

**Pull request merged**